nixos/tests/zrepl.nix

```diff
@@ -1,76 +1,179 @@
-                listen = ":9811";
-              }
-            ];
-            # Create a periodic snapshot job for an ephemeral zpool.
-            jobs = [
-              {
-                name = "snap_test";
-                type = "snap";
-
-                filesystems."test" = true;
-                snapshotting = {
-                  type = "periodic";
-                  prefix = "zrepl_";
-                  interval = "1s";
-                };
-
-                pruning.keep = [
-                  {
-                    type = "last_n";
-                    count = 8;
-                  }
-                ];
-              }
-            ];
+  nodes = {
+    source =
+      { nodes, pkgs, ... }:
+      {
+        config = {
+          # Prerequisites for ZFS and tests.
+          virtualisation.emptyDiskImages = [
+            2048
+          ];
+
+          boot.supportedFilesystems = [ "zfs" ];
+          environment.systemPackages = [
+            pkgs.parted
+            pkgs.zrepl
+          ];
+          networking.firewall.allowedTCPPorts = [ 8888 ];
+          networking.hostId = "deadbeef";
+          services.zrepl = {
+            enable = true;
+            settings = {
+              # Enable Prometheus output for status assertions.
+              global.monitoring = [
+                {
+                  type = "prometheus";
+                  listen = ":9811";
+                }
+              ];
+              # Create a periodic snapshot job for an ephemeral zpool.
+              jobs = [
+                {
+                  name = "snapshots";
+                  type = "snap";
+
+                  filesystems."tank/data" = true;
+                  snapshotting = {
+                    type = "periodic";
+                    prefix = "zrepl_";
+                    interval = "10s";
+                  };
+
+                  pruning.keep = [
+                    {
+                      type = "last_n";
+                      count = 8;
+                    }
+                  ];
+                }
+                {
+                  name = "backup-target";
+                  type = "source";
+
+                  serve = {
+                    type = "tcp";
+                    listen = ":8888";
+
+                    clients = {
+                      "${nodes.target.config.networking.primaryIPAddress}" = "${nodes.target.config.networking.hostName}";
+                      "${nodes.target.config.networking.primaryIPv6Address}" = "${nodes.target.config.networking.hostName
+                      }";
+                    };
+                  };
+                  filesystems."tank/data" = true;
+                  # Snapshots are handled by the separate snap job
+                  snapshotting = {
+                    type = "manual";
+                  };
+                }
+              ];
+            };
           };
         };
       };
-    };
+
+    target =
+      { pkgs, ... }:
+      {
+        config = {
```

Comment:
Running this test, I get:

```
evaluation warning: Module argument `nodes.target.config` is deprecated. Use `nodes.target` instead.
```

(it doesn't complain about `nodes.source.config` though lol?)